scripts/Imaging/imgpy/filters/rotate_stack.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import (absolute_import, division, print_function)
+import numpy as np
+import helper as h
+
+
+def cli_register(parser):
+    parser.add_argument(
+        "-r",
+        "--rotation",
+        required=False,
+        type=int,
+        help="Rotate images by 90 degrees a number of times.\n"
+        "The rotation is clockwise unless a negative number is given which indicates "
+        "rotation counterclockwise.")
+
+    return parser
+
+
+def gui_register(par):
+    raise NotImplementedError("GUI doesn't exist yet")
+
+
+def _rotate_image_inplace(data, rotation=None):
+    # rot90 rotates counterclockwise; config.pre.rotation rotates clockwise
+    data[:, :] = np.rot90(data[:, :], rotation)
+
+
+def _rotate_image(data, rotation=None):
+    # rot90 rotates counterclockwise; config.pre.rotation rotates clockwise
+    return np.rot90(data[:, :], rotation)
+
+
+def execute(data, rotation, flat=None, dark=None, cores=None, chunksize=None):
+    """
+    Rotates a stack (sample, flat and dark images).
+
+    This function only works with square images.
+
+    If the picture is cropped first, the ROI coordinates
+    have to be adjusted separately to be pointing at the NON ROTATED image!
+
+    :param data: stack of sample images
+    :param rotation: The rotation to be performed
+    :param flat: flat images average
+    :param dark: dark images average
+    :param cores: cores for parallel execution
+    :param chunksize: chunk for each worker
+
+    :return: rotated images
+    """
+    h.check_data_stack(data)
+    if rotation:
+        clockwise_rotations = 4 - rotation
+
+        from parallel import utility as pu
+        if pu.multiprocessing_available():
+            _execute_par(data, clockwise_rotations, cores, chunksize)
+        else:
+            _execute_seq(data, clockwise_rotations)
+
+        if flat is not None:
```

Comment:
Doesn't work with `numpy.ndarray`, `.any()` or `.all()` needs to be specified, and I am not sure how expensive that is for large arrays.